From 2c2117e4403ce2e51f4e24bd67c19f8433ac7080 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Mon, 28 Jan 2019 13:51:41 -0800
Subject: [PATCH 03/16] Fixups for cmake library configuration

- Install cmake config-file package to proper location
- Install library export file (targets.cmake)
- Restore IE:: namespacing to fix use of the config-file package
- Fix issue with build-time vs install-time header locations
---
 inference-engine/cmake/config.cmake.in             | 12 ++----------
 .../src/inference_engine/CMakeLists.txt            | 14 ++++++++++----
 2 files changed, 12 insertions(+), 14 deletions(-)

diff --git a/inference-engine/cmake/config.cmake.in b/inference-engine/cmake/config.cmake.in
index b32d288..7c3459f 100644
--- a/inference-engine/cmake/config.cmake.in
+++ b/inference-engine/cmake/config.cmake.in
@@ -5,17 +5,9 @@
 
 if(DEFINED IE_MAIN_SOURCE_DIR AND TARGET inference_engine)
     set(InferenceEngine_INCLUDE_DIRS ${IE_MAIN_SOURCE_DIR}/include)
-    if(NOT(UNIX))
-        set(InferenceEngine_LIBRARIES IE::inference_engine)
-    else()
-        set(InferenceEngine_LIBRARIES inference_engine)
-    endif()
+    set(InferenceEngine_LIBRARIES inference_engine)
 else()
     include("${CMAKE_CURRENT_LIST_DIR}/targets.cmake")
     get_target_property(InferenceEngine_INCLUDE_DIRS IE::inference_engine INTERFACE_INCLUDE_DIRECTORIES)
-    if(NOT(UNIX))
-        set(InferenceEngine_LIBRARIES IE::inference_engine)
-    else()
-        set(InferenceEngine_LIBRARIES inference_engine)
-    endif()
+    set(InferenceEngine_LIBRARIES IE::inference_engine)
 endif()
diff --git a/inference-engine/src/inference_engine/CMakeLists.txt b/inference-engine/src/inference_engine/CMakeLists.txt
index 8c18f0b..4a1e3c8 100644
--- a/inference-engine/src/inference_engine/CMakeLists.txt
+++ b/inference-engine/src/inference_engine/CMakeLists.txt
@@ -77,7 +77,9 @@ set_ie_threading_interface_for(${TARGET_NAME})
 target_link_libraries(${TARGET_NAME} PRIVATE fluid ade ${INTEL_ITT_LIBS} PUBLIC pugixml ${CMAKE_DL_LIBS})
 
 # Properties->C/C++->General->Additional Include Directories
-target_include_directories(${TARGET_NAME} PUBLIC ${PUBLIC_HEADERS_DIR}
+target_include_directories(${TARGET_NAME} PUBLIC
+                                          $<BUILD_INTERFACE:${IE_MAIN_SOURCE_DIR}/include>
+                                          $<INSTALL_INTERFACE:include/${TARGET_NAME}>
                                           PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}"
                                                   "${IE_MAIN_SOURCE_DIR}/src/dumper")
 
@@ -99,7 +101,9 @@ add_library(${TARGET_NAME}_s STATIC
 set_ie_threading_interface_for(${TARGET_NAME}_s)
 
 # Properties->C/C++->General->Additional Include Directories
-target_include_directories(${TARGET_NAME}_s PUBLIC ${PUBLIC_HEADERS_DIR}
+target_include_directories(${TARGET_NAME}_s PUBLIC
+                                          $<BUILD_INTERFACE:${IE_MAIN_SOURCE_DIR}/include>
+                                          $<INSTALL_INTERFACE:include/${TARGET_NAME}>
                                           PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}"
                                                   "${IE_MAIN_SOURCE_DIR}/src/dumper")
 
@@ -133,9 +137,11 @@ configure_file(
 
 add_cpplint_target(${TARGET_NAME}_cpplint FOR_TARGETS ${TARGET_NAME})
 
-install(TARGETS ${TARGET_NAME} LIBRARY DESTINATION lib64)
+set(ConfigPackageLocation lib64/cmake/InferenceEngine)
+install(TARGETS ${TARGET_NAME} EXPORT ${TARGET_NAME} LIBRARY DESTINATION lib64)
+install(EXPORT ${TARGET_NAME} DESTINATION ${ConfigPackageLocation} NAMESPACE IE:: FILE targets.cmake)
 install(FILES ${PUBLIC_HEADERS} DESTINATION include/${TARGET_NAME})
 install(FILES
     "${CMAKE_BINARY_DIR}/InferenceEngineConfig.cmake"
     "${CMAKE_BINARY_DIR}/InferenceEngineConfig-version.cmake"
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/share/cmake/Modules)
+    DESTINATION ${ConfigPackageLocation})
-- 
2.21.0

